generators/upgrade/index.js

```diff
@@ -180,10 +190,48 @@ module.exports = class extends BaseGenerator {
                 });
             },
 
-            checkLatestVersion() {
-                if (this.targetVersion) {
-                    this.log(`Upgrading to the target version: ${this.targetVersion}`);
-                    this.latestVersion = this.targetVersion;
+            checkLatestBlueprintVersion() {
```

Comment:
I was wondering instead of doing specific methods for generator and blueprint cant we generalize the methods to check and upgrade an npm lib and then pass the variables to that instead of duplicating code

Reply:
So here this method can just call another method which does the check twice, first for package `generator-jhipster` and then for package `generator-${blueprint}`

Reply:
I've done a bit of refactoring, please let me know what you think.